lib/mempool/addrindexer.js

```diff
@@ -0,0 +1,235 @@
+/*!
+ * mempool.js - mempool for bcoin
+ * Copyright (c) 2018-2019, the bcoin developers (MIT License).
+ * Copyright (c) 2014-2017, Christopher Jeffrey (MIT License).
+ * https://github.com/bcoin-org/bcoin
+ */
+
+'use strict';
+
+const assert = require('bsert');
+const {BufferMap} = require('buffer-map');
+const TXMeta = require('../primitives/txmeta');
+
+/**
+ * Address Indexer
+ * @ignore
+ */
+
+class AddrIndexer {
+  /**
+   * Create TX address index.
+   * @constructor
+   */
+
+  constructor() {
+    // Map of addr->entries.
+    this.index = new BufferMap();
+
+    // Map of txid->addrs.
+    this.map = new BufferMap();
+  }
+
+  reset() {
+    this.index.clear();
+    this.map.clear();
+  }
+
+  getKey(addr) {
+    const prefix = addr.getPrefix();
+
+    if (prefix < 0)
+      return null;
+
+    const raw = Buffer.allocUnsafe(1);
+    raw.writeUInt8(prefix);
+
+    return Buffer.concat([raw, addr.getHash()]);
```

Comment:
Instead of Buffer concat (which is another allocation), you could do it in one go

e.g.
```javascript
    const hash = addr.getHash();
    const size = 1 + hash.length;
    const raw = Buffer.allocUnsafe(size);

    let written = raw.writeInt8(n);
    written += hash.copy(raw, 1);
    assert(written === size);

   return raw;
```

Reply:
Worth considering, measurable performance improvement?

Reply:
almost2x in simple benchmarks

Reply:
*more than 2x